sdk/state/ingest.go

```diff
@@ -15,71 +15,70 @@ import (
 // determine if successful or not, and understand changes in the ledger as a
 // result.
 func (c *Channel) IngestTx(tx *txnbuild.Transaction) error {
-	// Use the tx to update the latest unauthorized close agreement if possible.
-	err := c.ingestTxToUpdateUnauthorizedCloseAgreement(tx)
+	// TODO: Use the transaction result to affect on success/failure.
+
+	c.ingestTxToUpdateInitiatorEscrowAccountSequence(tx)
+
+	// If we found an unauthorized close agreement has begun closing, update our unauthorized to
+	// become authorized.
+	closeState, err := c.CloseState()
 	if err != nil {
-		return err
+		return fmt.Errorf("getting close state: %w", err)
 	}
 
-	// TODO: If the tx hash matches an authorized or unauthorized declaration,
-	// mark the channel as closing.
-	// TODO: If the tx hash matches an authorized or unauthorized close, mark the
-	// channel as closed.
-	// TODO: If the tx is for an older declaration, mark the channel as closing with
-	// requiring bump.
-	// TODO: Use the transaction result to affect on success/failure.
+	if closeState == CloseStateEarlyClosing {
+		err := c.updateUnauthorizedAgreement(tx)
+		if err != nil {
+			return err
+		}
+		closeState, err := c.CloseState()
+		if err != nil {
+			return fmt.Errorf("getting close state: %w", err)
+		}
+		if closeState != CloseStateClosing {
+			return fmt.Errorf("converting unauthorized agreement to authorized should put channel in closing state")
+		}
+	}
```

Comment:
gotchya, my thinking was only call `updateUnauthorizedAgreement` when needed, so using the `CloseStateEarlyClosing` state. Calling it every time makes sense to me I should have changed it to that. Changed back to here: https://github.com/stellar/experimental-payment-channels/pull/232/commits/ec6870bb18fe64951de8df8800bb495699a072f4

which has the original code of checking transaction hashes, returning nil if not relevant etc. Also removes the `CloseStateEarlyClosing` since we don't need it anymore